vpt/processors/gaze_engagement_estimator/__init__.py

```diff
@@ -10,9 +10,16 @@ class GazeEngagementEstimator(ProcessorBase[np.ndarray]):
     '''Transforms the gaze estimation to engagement estimation.'''
     subj = Subject()
 
-    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray]):
+    def __init__(self, rotation_vector_source: ProcessorBase[np.ndarray], threshold=0.5):
+        self.threshold = threshold
         rotation_vector_source.get_data_stream().subscribe(lambda v:
-                                                             self.subj.on_next(np.linalg.norm(v)))
+                                                           self.subj.on_next(np.linalg.norm(v) < self.threshold))
+
+    def start(self):
+        pass
```

Comment:
Ah, yes... Processors are indeed sources...

Reply:
What I don't understand is why isn't the default implementation for `start` in Processors just `self.subj.start()`

Reply:
`subj` does not have any start methods, think of it as of UNIX pipe. Can you start a UNIX pipe?

Reply:
I probably meant the sources it depends on that are passed in the constructor, but then I realized that you cannot do that in the default implementation in the parent class.
But still, all `Processor` subclasses should override `start` to call `.start()` on all sources they depend on, not just `pass`

Reply:
It sounds like a good solution, but let's not bother with this and just start all sources manually, please. This will bring no value to our project

Reply:
Sure, but we're currently not even doing that. Most of the processors are currently black holes (they will not call `start` on their own sources) or even outright throw an exception.

Reply:
Processors are not expected to call `start` on their sources, the way we currently have it. We just assume that a source will come bearing data

Reply:
yeah, well, that shouldn't be the case. It doesn't make much sense. From the perspective of sinks, processors are just sources, and calling `start` on sources should \[surprisingly enough\] _start_ them. How can they start? By asking their own sources to start so that they will actually have data

Reply:
We don't need to call `start()` on processors. The reason why we have it is because of some bullshit with abstract classes that requires them